app/utils/timeHelpers.ts

```diff
@@ -267,29 +274,71 @@ export const isDateEqual = (
     return stripTime(left).getTime() === stripTime(right).getTime();
 };
 
+/**
+ * Gets time of the payday after a given time.
+ *
+ * @param time the time for which the succeeding payday is returned
+ * @param nextPayday the next payday in relation to current time
+ * @param rewardPeriodLengthMS the length of a payday in milliseconds
+ */
+export const getSucceedingPayday = (
```

Comment:
what is this :thinking:

Reply:
nvm, I think I get it. You are finding the payday after the effective time by recursive, linear search?

Reply:
Yes! 🥇